pkgs/development/tools/misc/gdbgui/default.nix

```diff
@@ -0,0 +1,40 @@
+{ stdenv, python27Packages, gdb, pkgs }:
+let
+  deps = import ./requirements.nix { inherit pkgs; };
+in
+python27Packages.buildPythonApplication rec {
+    name = "${pname}-${version}";
+    pname = "gdbgui";
+    version = "0.9.0.1";
+
+    buildInputs = [ gdb ];
+    propagatedBuildInputs = builtins.attrValues deps.packages;
+
+    src = python27Packages.fetchPypi {
+      inherit pname version;
+      sha256 = "1gjc7dycrc4zafhrd9yib7qnh4agh7cpa6rlw4p5405rlmwmsbj3";
+    };
+
+    postInstall = ''
+      wrapProgram $out/bin/gdbgui \
+        --prefix PATH : ${stdenv.lib.makeBinPath [ gdb ]}
+    '';
+
+    # make /etc/protocols accessible to fix socket.getprotobyname('tcp') in sandbox
+    preCheck = ''
+      export NIX_REDIRECTS=/etc/protocols=${pkgs.iana-etc}/etc/protocols \
+             LD_PRELOAD=${pkgs.libredirect}/lib/libredirect.so
+    '';
```

Comment:
This fixes tests for me as well. The only disadvantage I see that this method is not available on darwin (and maybe not required?). Can you make this optional on linux:

```
preCheck = lib.optionalString stdenv.isLinux ''
      export NIX_REDIRECTS=/etc/protocols=${pkgs.iana-etc}/etc/protocols \
             LD_PRELOAD=${pkgs.libredirect}/lib/libredirect.so
'';
```

then we can check with our pr bot, if it builds on darwin.

Reply:
Thanks, amended!